### PR TITLE
feat(router): support optional route

### PR DESCRIPTION
**Pull request by yusukebe**
This PR introduces "optional parameter":

```ts
app.get('/api/animals/:type?',() => {})
```

Will match:

```http
GET /api/animals
GET /api/animals/dog
```

Reference:
https://router.vuejs.org/guide/essentials/route-matching-syntax.html#sensitive-and-strict-route-options

### CODE REVIEW

**Comment by yusukebe** [comment]
@usualoma !

If you agree on this specification and implementation, could you make it for RegExpRouter? That it wouldn't be hard to follow this implementation.

**Comment by usualoma** [comment]
@yusukebe 
How about #542?

**Comment by yusukebe** [comment]
Hi @usualoma !

Looks good! I will merge it.
